finance_dl/discover.py

```diff
@@ -0,0 +1,176 @@
+"""Retrieves transaction and balance information from Discover.
+Logs into the web interface, downloads an OFX file.
+
+This uses the `selenium` Python package in conjunction with `chromedriver` to
+scrape the Discover website.
+
+Configuration:
+==============
+
+The following keys may be specified as part of the configuration dict:
+
+- `credentials`: Required.  Must be a `dict` with `'username'` and `'password'`
+  keys.
+
+- `output_directory`: Required.  Must be a `str` that specifies the path on the
+  local filesystem where the output will be written.  If the directory does not
+  exist, it will be created.
+
+- `profile_dir`: Optional.  If specified, must be a `str` that specifies the
+  path to a persistent Chrome browser profile to use.  This should be a path
+  used solely for this single configuration; it should not refer to your normal
+  browser profile.  If not specified, a fresh temporary profile will be used
+  each time.  It is highly recommended to specify a `profile_dir` to avoid
+  having to manually enter a multi-factor authentication code each time.
+
+Output format:
+==============
+
+OFX file.
+
+Example:
+========
+
+    def CONFIG_discover():
+        return dict(
+            module='finance_dl.discover',
+            credentials={
+                'username': 'XXXXXX',
+                'password': 'XXXXXX',
+            },
+            output_directory=os.path.join(data_dir, 'discover'),
+            # firefox seems to work better on Discover's website
+            profile_dir=os.path.join(profile_dir, 'firefox'),
```

Comment:
Maybe include `firefox=True` as well here.

Reply:
After some more testing I've concluded that Chrome is preferable because it does the job faster, and Firefox requires an extra config step to tell it to auto-save OFX files. A year ago when I wrote this code Firefox was the only way.
Given that it is not necessary for Discover, is it worth keeping the code extending scrape_lib (Firefox as supported driver)?

Reply:
Seems reasonable to leave out the firefox support then, I suppose we can always find it in this pull request later if desired

Reply:
Done.